tests/test_workflow_api.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+from devtools import debug
+from sqlmodel import select
+
+from fractal_server.app.models import Task
+from fractal_server.app.models import TaskCreate
+from fractal_server.tasks import collect_tasks
+
+async def test_workflow_post(db, client, MockCurrentUser):
+    fake_workflow = {
+        "name": "My Workflow",
+        "resource_type": "core task",
+        "input_type": "Boh",
+        "output_type": "Boh",
```

Comment:
these parameters are not part of the Workflow schema, cf. `fractal_server/app/schames/task.py` and `fractal_server/app/models/workflow.py`

in particular, we should test that if a workflow is defined without providing a project id, the request fails.

Reply:
If I do
```
res = await client.post(
            "api/v1/workflow/",
            json={
                "name": "My Workflow",
                "project_id": 1,
            },
)
debug(res.json())
```
I get
```
res.json(): {
        'detail': [
            {
                'loc': [
                    'body',
                    'resource_type',
                ],
                'msg': 'field required',
                'type': 'value_error.missing',
            },
            {
                'loc': [
                    'body',
                    'input_type',
                ],
                'msg': 'field required',
                'type': 'value_error.missing',
            },
            {
                'loc': [
                    'body',
                    'output_type',
                ],
                'msg': 'field required',
                'type': 'value_error.missing',
            },
        ],
    }
```
Why does it require these fields if they are not part of Workflow?